### PR TITLE
refactor(tests): migrate to mocha from jasmine

### PR DESCRIPTION
**Pull request by saiponnada**

### CODE REVIEW

**Comment by ianmcburnie** [comment]
So it appears we still have issues with false positives. Try changing the expected value here and see that it always passes:

https://github.com/makeup/makeup-js/blob/90e5fe0b5823489269a39bcdaaa4538c00bb1da5/packages/makeup-navigation-emitter/test/index.js#L735

When I remove the timeout, and change the expected value, the test fails as expected. So maybe we need to go through and remove all timeouts now?

**Comment by ianmcburnie** [comment]
@agliga recommended we look into testing-library:

https://testing-library.com/docs/dom-testing-library/intro
https://testing-library.com/docs/user-event/intro

File: packages/makeup-exit-emitter/test/index.js [hunk 1]
```diff
@@ -21,33 +23,58 @@ describe('given an element with focus', function() {
         testElSibling = document.querySelector('#test-element-sibling');
         ExitEmitter.addFocusExit(testEl);
         testEl.addEventListener('onFocusExit', onFocusExit);
```
Comment (saiponnada) [inline packages/makeup-exit-emitter/test/index.js:25]: @ianmcburnie , I guess this should be 
`testEl.addEventListener('focusExit', onFocusExit);` thats why failing most of the tests..

**Comment by ianmcburnie** [comment]
Before looking into testing-library, let's just try removing all the timeouts from what you currently have and see how that fares.

**Review by ianmcburnie — APPROVED** [review]
Great set of changes @saiponnada ! I will merge.

**Pull request merged**